pyquil/api/_compiler.py

```diff
@@ -106,10 +108,23 @@ def native_quil_to_executable(
 
         If `api_options` is provided, it overrides the options set on `self`.
         """
-        rewrite_response = rewrite_arithmetic(nq_program.out())
+        program = nq_program.out()
+        recalculation_table = []
+        # NOTE: There is an incompatibility between backends v1 and v2: v1 expects
+        # arithmetic to have been rewritten by the client, while v2 expects
+        # arithmetic _not_ to have been rewritten by the client. Since Aspen-M-3
+        # is the last of our QPUs that is compatible withbackend v1, we only rewrite arithmetic
+        # for that particular QPU. Once Aspen-M-3 is EOL, we can remove this.
```

Comment:
This seems to belong on the validate function, since it describes its implementation and purpose.